lib/utils/useWeb3Modal/index.tsx

```diff
@@ -1,141 +1,129 @@
+      await web3state.provider?.provider?.close();
     }
+    window.location.reload();
   };
 
   const connect = async (wallet?: string): Promise<void> => {
     const connectedWallet = localStorage.getItem("web3-wallet");
     try {
+      let provider: TypedProvider;
+
+      /** HANDLE METAMASK / INJECTED */
       if (
         wallet === "metamask" ||
         wallet === "brave" ||
         connectedWallet === "injected"
       ) {
-        const provider = new ethers.providers.Web3Provider(
-          window.ethereum as any
-        ) as any;
+        provider = getWeb3Provider(window.ethereum);
         // if user is not already connected this request will prompt the wallet modal to open and the user to connect
         await provider.send("eth_requestAccounts", []);
-        const signer = provider.getSigner();
-        const address = await signer.getAddress();
-        const network = await provider.getNetwork();
-        const newState: ConnectedWeb3State = {
-          provider,
-          signer,
-          address,
-          network,
-          isConnected: true,
-        };
-        setWeb3State(newState);
         localStorage.setItem("web3-wallet", "injected");
       } else if (wallet === "coinbase" || connectedWallet === "coinbase") {
-        const CoinbaseWalletSDK = (await import("@coinbase/wallet-sdk"))
-          .CoinbaseWalletSDK;
-        const coinbaseWallet = new CoinbaseWalletSDK({
+        const { default: CoinbaseWalletSDK } = await import(
+          "@coinbase/wallet-sdk"
+        );
+        const { makeWeb3Provider } = new CoinbaseWalletSDK({
           appName: "KlimaDAO App",
-          darkMode: false,
+          darkMode: false, // TODO: get theme from body/localstorage
         });
-
-        const provider = new ethers.providers.Web3Provider(
-          coinbaseWallet.makeWeb3Provider(urls.polygonMainnetRpc, 137) as any
-        ) as unknown as TypedProvider;
+        provider = getWeb3Provider(
+          makeWeb3Provider(urls.polygonMainnetRpc, 137)
+        );
         // if user is not already connected this request will prompt the wallet modal to open and the user to connect
         await provider.send("eth_requestAccounts", []);
-        const signer = await provider.getSigner();
-        const address = await signer.getAddress();
-        const network = await provider.getNetwork();
-        const newState: ConnectedWeb3State = {
-          provider,
-          signer,
-          address,
-          network,
-          isConnected: true,
-        };
-        setWeb3State(newState);
         localStorage.setItem("web3-wallet", "coinbase");
+
+        /** HANDLE WALLETCONNECT */
       } else if (
         wallet === "walletConnect" ||
         connectedWallet === "walletConnect"
       ) {
-        const WalletConnectProvider = (
-          await import("@walletconnect/web3-provider")
-        ).default;
+        const { default: WalletConnectProvider } = await import(
```

Comment:
@goatsOnDrugs this is just a personal preference for me, something about that extra `( ).default` annoys me